src/controller/CHIPDeviceController.cpp

```diff
@@ -118,14 +118,22 @@ CHIP_ERROR DeviceController::Init(ControllerInitParams params)
     mDNSResolver.SetCommissioningDelegate(this);
     RegisterDeviceDiscoveryDelegate(params.deviceDiscoveryDelegate);
 
-    VerifyOrReturnError(params.operationalCredentialsDelegate != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
-    mOperationalCredentialsDelegate = params.operationalCredentialsDelegate;
-
     mVendorId = params.controllerVendorId;
     if (params.operationalKeypair != nullptr || !params.controllerNOC.empty() || !params.controllerRCAC.empty())
     {
         ReturnErrorOnFailure(InitControllerNOCChain(params));
     }
+    else if (params.enableUseStorageFabric && params.useStorageFabricIndex != kUndefinedFabricIndex)
+    {
+        VerifyOrReturnError(params.systemState->Fabrics()->FabricCount() >= kMinValidFabricIndex, CHIP_ERROR_INVALID_ARGUMENT);
+        const FabricInfo * fabricInfo = nullptr;
+        fabricInfo                    = params.systemState->Fabrics()->FindFabricWithIndex(params.useStorageFabricIndex);
+        if (fabricInfo != nullptr)
+        {
+            // ensure the fabric is existed
+            mFabricIndex = fabricInfo->GetFabricIndex();
```

Comment:
is this the same as `params.useStorageFabricIndex` or could it be something else?

Is this code maybe: 

```cpp
if (params.systemState->Fabrics()->FindFabricWithIndex(params.useStorageFabricIndex) != nullptr)
{
    // valid index, can be used
    mFabricIndex = params.useStorageFabricIndex;
}
else
{
   // add log here saying this index does not exist
}
```

Reply:
If the fabric is found, the FabricIndex is same. Have done.